openc3-cosmos-script-runner-api/app/controllers/running_script_controller.rb

```diff
@@ -50,21 +50,60 @@ def stop
   end
 
   def delete
-    running_script = RunningScript.find(params[:id].to_i)
+    running_script = OpenC3::ScriptStatusModel.get_model(name: params[:id], scope: params[:scope])
     if running_script
-      target_name = running_script['name'].split('/')[0]
+      target_name = running_script.name.split('/')[0]
+      pid = running_script.pid.to_i
       return unless authorization('script_run', target_name: target_name)
       running_script_publish("cmd-running-script-channel:#{params[:id]}", "stop")
-      RunningScript.delete(params[:id].to_i)
-      OpenC3::Logger.info("Script deleted: #{running_script}", scope: params[:scope], user: username())
+
+      # Give the process 1 second to stop from stop message
+      stopped = false
+      start_time = Time.now
+      while Time.now - start_time < 1.0
+        begin
+          Process.getpgid(pid.to_i)
+          sleep 0.1
+        rescue Errno::ESRCH
+          stopped = true
+          break
+        end
+      end
+
+      # If the process is still running
+      # Send a SIGINT and give it one more second
+      if not stopped
+        Process.kill("SIGINT", pid)
+        start_time = Time.now
+        while Time.now - start_time < 1.0
+          begin
+            Process.getpgid(pid.to_i)
+            sleep 0.1
+          rescue Errno::ESRCH
+            stopped = true
+            break
+          end
+        end
+      end
+
+      # If the process is still running send a hard kill signal
+      if not stopped
+        Process.kill("SIGKILL", pid)
+
+        # Also need to cleanup the status model in this case because the process will not die and cleanup
+        running_script.state = "killed"
+        running_script.update
+      end
```

Comment:
Were we just leaving zombie processes before with the `RunningScript.delete` by itself?

Reply:
Potentially.  Though most of the time I think they were not really there due to a container restart.